### ISSUE TITLE
 Stopping a process is broken.

### ISSUE THREAD

**Issue by stijnveenman** [F1bonacc1/process-compose#149]
#### Version of `process-compose`:
```
Process Compose
Version:        v0.88.0
Commit:         1a73f44
Date (UTC):     2024-03-02T23:44:10Z
License:        Apache-2.0
Discord:        https://discord.gg/S4xgmRSHdC

Written by Eugene Berger
```
#### OS environment:
Mac M2

#### Steps or code to reproduce the issue:
With the main `process-compose.yaml` in this repo run
`process-compose up _process2 process4-0`
Stop (F9) `_process2` 

#### Expected result:
When stopping a process, i expect it to stop

#### Actual result:
The process is stopped and restarted. 

I thought this might be related to 
```
    availability:
      restart: "on_failure"
```
as we have this in our configuration. But removing this on `_process2` does not seem to resolve the issue

**Comment by F1bonacc1** [comment]
Hey @stijnveenman,

A few questions/comments:
1. There is the `porcess-compose.overrides.yaml` file, did you remove the `on_failure` parameter from it as well?
2. If you use test_loop.bash, its trap is configured to exit with the `$EXIT_CODE` configured for the process. It's non 0 for the processes you mentioned.

**Comment by stijnveenman** [comment]
Thanks for the quick response!

1. Ahh, i did check `process-compose.overrides.yaml` but must have read over it, removing it from here does indeed prevent the restarts. 
2. I changed the env variable of `EXIT_CODE` to 0 and this also fixed the issue. 

So the behaviour seems to be, when stopping a process results in a non 0 exit code, and `restart: on_failure` is set it will be restarted. This somewhat makes sense

I am of the opinion that this should however still result in the process being stopped and not restarted if it's stopped manually. Though we can just remove the `on_failure` from our config otherwise.

**Comment by F1bonacc1** [comment]
I guess the behavior that you want is "restart on failure until stopped manually".
This is indeed not supported today.

I think I can add it.
In the following scenario:
1. A process is configured to restart on failure or restart always
2. Manual Stop
3. Manual Start
Should it go back to the previous (on failure, always) behavior or run until completion?

**Comment by thenonameguy** [comment]
Just +1'ing this issue as I've been hit by it as well. TUI/CLI start/stop intents should always ignore restart policies, or at least return a warning (in case of CLI).

**Comment by stijnveenman** [comment]
I agree with @thenonameguy 

Regarding your scenario, yes i think it should go back to the previous restart mode. So any TUI action ignores the restart policy, but if started again it goes back to the normal configuration

**Comment by F1bonacc1** [comment]
Agreed.
I will add it to the next release.

**Comment by F1bonacc1** [comment]
Fixed in v1.0.0
